File: backend/app/api/roadmap.py
from fastapi import APIRouter
from typing import Optional
import uuid

router = APIRouter()

@router.get("/")
async def get_roadmap():
    return {
        "phases": [],
        "summary": {"completed": 0, "in_progress": 0, "pending": 0}
    }

@router.post("/{id}/start")
async def start_roadmap_phase(id: str):
    job_id = str(uuid.uuid4())
    return {"job_id": job_id, "message": "Phase started"}

File: frontend/src/api/roadmap.ts
import client from './client';
import type { RoadmapPhase } from '../types';

export interface RoadmapSummary {
  completed: number;
  in_progress: number;
  pending: number;
}

export const getRoadmap = async (): Promise<{ phases: RoadmapPhase[], summary: RoadmapSummary }> => {
  const response = await client.get('/roadmap');
  return response.data;
};

export const startRoadmapPhase = async (phaseId: string): Promise<{ job_id: string, message: string }> => {
  const response = await client.post(`/roadmap/${phaseId}/start`);
  return response.data;
};
